src/Npgsql/PostgresDatabaseInfo.cs

```diff
@@ -383,63 +359,74 @@ static void LoadCompositeFields(NpgsqlDataReader reader, Dictionary<uint, Postgr
-                    currentOID = oid;
+                // Then load the enum fields
+                Expect<RowDescriptionMessage>(await conn.ReadMessage(async), conn);
 
-                    if (!byOID.TryGetValue(oid, out var type))  // See #2020
+                currentOID = uint.MaxValue;
+                PostgresEnumType? currentEnum = null;
+                skipCurrent = false;
+
+                while (true)
+                {
+                    msg = await conn.ReadMessage(async);
+                    if (msg is not DataRowMessage)
+                        break;
+
+                    buf.Skip(2); // Column count
+                    var oid = uint.Parse(ReadNonNullableString(buf), NumberFormatInfo.InvariantInfo);
+                    var enumlabel = ReadNonNullableString(buf);
+                    if (oid != currentOID)
                     {
-                        Log.Warn($"Skipping enum type with OID {oid} which was not found in pg_type");
-                        byOID.Remove(oid);
-                        skipCurrent = true;
-                        continue;
+                        currentOID = oid;
+
+                        if (!byOID.TryGetValue(oid, out var type))  // See #2020
+                        {
+                            Log.Warn($"Skipping enum type with OID {oid} which was not found in pg_type");
+                            byOID.Remove(oid);
+                            skipCurrent = true;
+                            continue;
+                        }
+
+                        currentEnum = type as PostgresEnumType;
+                        if (currentEnum == null)
+                        {
+                            Log.Warn($"Type {type.Name} was referenced as an enum type but is a {type.GetType()}");
+                            byOID.Remove(oid);
+                            skipCurrent = true;
+                            continue;
+                        }
+
+                        skipCurrent = false;
                     }
 
-                    currentEnum = type as PostgresEnumType;
-                    if (currentEnum == null)
-                    {
-                        Log.Warn($"Type {type.Name} was referenced as an enum type but is a {type.GetType()}");
-                        byOID.Remove(oid);
-                        skipCurrent = true;
+                    if (skipCurrent)
                         continue;
-                    }
 
-                    skipCurrent = false;
+                    currentEnum!.MutableLabels.Add(enumlabel);
                 }
+                Expect<CommandCompleteMessage>(msg, conn);
+            }
 
-                if (skipCurrent)
-                    continue;
+            Expect<ReadyForQueryMessage>(await conn.ReadMessage(async), conn);
+            return byOID.Values.ToList();
 
-                currentEnum!.MutableLabels.Add(reader.GetString("enumlabel"));
-            }
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
```

Comment:
Should not be necessary (both because the JIT will do it anyway, and also because this is a very cold method), up to you.

Reply:
Yeah. I kind of do it on principle for tiny helpers which probably is a bad habit because tiny methods will get inlined anyway and for bigger ones it's probably better to leave the decision to the jitter.
I'll remove it.